torch/csrc/jit/passes/utils/subgraph_utils.cpp

```diff
@@ -157,6 +200,13 @@ void mergeNodeIntoSubgraph(Node* toMerge, Node* subgraphNode) {
   auto mergedNode = subgraph->insertNode(
       subgraph->createClone(toMerge, [&](Value* v) { return inputsMap[v]; }));
 
+  for (size_t idx = 0; idx < toMerge->outputs().size(); idx++) {
+    vmap[toMerge->output(idx)] = mergedNode->output(idx);
+  }
+  for (size_t idx = 0; idx < toMerge->inputs().size(); idx++) {
+    vmap[toMerge->input(idx)] = mergedNode->input(idx);
+  }
+
   // If n's outputs were inputs to `group`, remove them since we just merged
```

Comment:
So, what happens in a case like this: https://github.com/pytorch/pytorch/blob/2c287a20c78b7b7f468f56bfa2a91d080520d709/torch/csrc/jit/passes/utils/subgraph_utils.cpp#L240 ?

If a node is merged into a subgraph, the vmap mapping contains only the mapping from the value to its new value within the subgraph, not its new value in the enclosing graph correct ?

Reply:
Correct, we don't have a mapping in the enclosing graph. So, we won't notice that a value have been added to inputs/outputs.

I actually thought about some API to detect that a value was registered as a new input/output, but couldn't come up with clean rules for doing so and instead went with the approach presented in this PR.